### PR TITLE
story/VOGRE-1

### PR DESCRIPTION
**Pull request by Girik1105**
Ticket Link: [VOGRE-1](https://diging.atlassian.net/browse/VOGRE-1?atlOrigin=eyJpIjoiNTFkODAxYjFlNjUzNDBiNjgyY2U3MjI0OGMzMzJhZjgiLCJwIjoiaiJ9)

In Profile page (accounts/profile/) I created the external accounts section where users can connect their accounts to Citesphere.

You will need to add the following Settings in your local env to run this branch:
CITESPHERE_ENDPOINT=
CITESPHERE_CLIENT_ID=
CITESPHERE_CLIENT_SECRET=
CITESPHERE_REDIRECT_URI=http://localhost:8000/oauth/callback/citesphere/

Additional Notes:
I merged VOGRE-2 in this branch to make authentication easier.

[VOGRE-1]: https://diging.atlassian.net/browse/VOGRE-1?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jdamerow** [comment]
Does this mean the other PR is obsolete? If so, please close it. also, unless absolutely necessary, two tickets should not be merged into their own branch. It gets really messy, really quickly. Local authentication should still work without the Citeresphere change, shouldn't it?

**Comment by jdamerow** [comment]
there should be a `.env-example` file that has a list of all environment variables that should be set.

**Comment by jdamerow** [comment]
It's not clear to me where a user would add citesphere.

**Comment by Girik1105** [comment]
> Does this mean the other PR is obsolete? If so, please close it. also, unless absolutely necessary, two tickets should not be merged into their own branch. It gets really messy, really quickly. Local authentication should still work without the Citeresphere change, shouldn't it?

I merged VOGRE-2 which is switching to local authentication, as I wasn't sure how would I switch from GitHub auth to local auth while enabling users to connect to citesphere, I'll close the PR for VOGRE-2 (local auth) as all the changes of the ticket is in this PR

**Comment by Girik1105** [comment]
> It's not clear to me where a user would add citesphere.

The user can go to his Dashboard page in the dropdown (http://0.0.0.0:8000/accounts/profile/)
Screenshot 2024-08-29 at 9 17 25 AM
Here the user would be able to see the option to connect to Citesphere.
Screenshot 2024-08-29 at 9 19 41 AM

**Comment by jdamerow** [comment]
That did not show up for me (but also, the layout is messed up for).

**Comment by jdamerow** [comment]
I'll try it again

**Comment by jdamerow** [comment]
> there should be a .env-example file that has a list of all environment variables that should be set.

This doesn't seem to be there?

**Comment by jdamerow** [comment]
still not working? the bower style sheets are still listed in `stylesheets.html` and bootstrap has not been added?

File: annotations/static/annotations/.DS_Store
Comment (jdamerow) [inline annotations/static/annotations/.DS_Store:1]: please remove and ignore

**Comment by jdamerow** [comment]
- `EMAIL_BACKEND` seems to be in the env example file twice.
- After connecting to Citesphere, it doesn't who that citesphere has been connected.

**Comment by jdamerow** [comment]
no citesphere account seems to be stored after connecting

**Comment by jdamerow** [comment]
ah! it's because the redirect link is incorrect. can you add example links into the example env please.

**Comment by jdamerow** [comment]
I'm going to merge this, but I get the following error when trying to import a collection:
`{"error": "time data '2012-01-01T00:00Z' does not match format '%Y-%m-%dT%H:%M:%SZ'"}`

and i have other comments, but I'll do that in VOGRE-4

**Pull request merged**